File: examples/jump/sudoku/sudoku_api.jl
using Pkg
Pkg.activate(".") 

import DotEnv
DotEnv.config()

using MOSInterface

# Interface
interface = Interface()

# Delete model
delete_model_with_name(interface, "Sudoku JuMP Model")

# New model
model = new_model(interface, "./examples/jump/sudoku/sudoku_model.jl")

# Get model
model = get_model_with_name(interface, "Sudoku JuMP Model")

# Set inputs
set_interface_file(model, "initial_grid_file", "./examples/jump/sudoku/data.csv")

# Show info
@assert(get_name(model) == "Sudoku JuMP Model")
@assert(get_system(model) == "jump")
@assert(get_status(model) == "created")

# Solve
MOSInterface.run(model)
@assert(get_status(model) == "success")

File: examples/jump/sudoku/sudoku_api.py
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

from mos.interface import Interface

# Interface
interface = Interface()

# Delte model
interface.delete_model_with_name('Sudoku JuMP Model')

# New model
model = interface.new_model('./examples/jump/sudoku/sudoku_model.jl')

# Existing model by name
model = interface.get_model_with_name('Sudoku JuMP Model')

# Set inputs
model.set_interface_file('initial_grid_file', 
                  './examples/jump/sudoku/data.csv')

# Solve 
model.run()
